### PR TITLE
Add 345 release notes

### PR DESCRIPTION
**Pull request by martint**
Closes https://github.com/prestosql/presto/issues/5503

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+* Add support for JSON Web Key (JWK) to the existing JSON Web Token (JWT) authenticator.  This is enabled by 
+  setting the `jwt.key-file` configuration property to a `http` or `https` url. ({issue}`5419`)
+* Add column security, column mask and row filter to file-based access controls. ({issue}`5460`)
+* Enforce access control for column references in `USING` clause. ({issue}`5620`)
+
+## Hive Connector Changes
+
+* Use `hive.domain-compaction-threshold` threshold for compacting dynamic filters. ({issue}`5365`)
+* Support reading Parquet timestamp encoded as microseconds. ({issue}`5483`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-345.md:26]: "Add support for"

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+## Security Changes
+
+* Add support for JSON Web Key (JWK) to the existing JSON Web Token (JWT) authenticator.  This is enabled by 
+  setting the `jwt.key-file` configuration property to a `http` or `https` url. ({issue}`5419`)
+* Add column security, column mask and row filter to file-based access controls. ({issue}`5460`)
+* Enforce access control for column references in `USING` clause. ({issue}`5620`)
+
+## Hive Connector Changes
+
+* Use `hive.domain-compaction-threshold` threshold for compacting dynamic filters. ({issue}`5365`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-345.md:25]: Maybe:

Apply `hive.domain-compaction-threshold` to dynamic filters

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+* Support reading Parquet timestamp encoded as microseconds. ({issue}`5483`)
+* Fix failure when reading Parquet `timestamp` columns encoded as `int64`. ({issue}`5443`)
+
+## MongoDB Connector Changes
+
+* Add support for adding a column. ({issue}`5512`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-345.md:31]: Maybe: `Add support for `ADD COLUMN`.  Not sure about this as the syntax is `ALTER TABLE name ADD COLUMN`

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+* Improve performance of queries with aggregations and `LIMIT` clause (but without `ORDER BY`). ({issue}`5261`)
+
+## PostgreSQL Connector Changes
+
+* Improve performance of queries with aggregations and `LIMIT` clause (but without `ORDER BY`). ({issue}`5261`)
+
+## Redshift Connector Changes
+
+* Add support for setting a column comment. ({issue}`5397`)
```
Comment (dain) [inline presto-docs/src/main/sphinx/release/release-345.md:43]: Maybe `Add support for setting column comments`

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+* Add {func}`concat_ws` function. ({issue}`4680`)
+* Add support for {func}`extract` for `time with time zone` values with precision other than 3. ({issue}`5539`)
+* Add dynamic filtering support for right joins. ({issue}`5555`) 
+* Add support for catalog session property to file based session property manager. ({issue}`5330`)
+* Add support for catalog session property to database based session property manager. ({issue}`5330`)
+* Improve query scalability by increasing broadcast join buffer size to 200MB.
+  Broadcast join buffer size can be set using `sink.max-broadcast-buffer-size` feature config. ({issue}`5551`)
+* Improve query performance by allowing larger dynamic filters. ({issue}`5529`)
+* Improve dynamic filtering across connectors by adding implicit cast support for dynamic filters. ({issue}`5461`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-345.md:13]: How does it sound from end user perspective?

maybe something about: improve performance for join queries where join keys are not of the same type (eg integer and bigint)

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+* Improve performance when encrypted spilling is enabled. ({issue}`5557`)
```
Comment (findepi) [inline presto-docs/src/main/sphinx/release/release-345.md:14]: "Improve performance of encrypted spilling"?

File: presto-docs/src/main/sphinx/release/release-345.md (hunk 1 above)
Comment (findepi) [reply presto-docs/src/main/sphinx/release/release-345.md:26]: i am fine with that. should i start/prefix every new feature with "Add support for .." then? @dain ?

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
+# Release 345 (XX Oct 2020)
+
+## General Changes
+
+* Add {func}`concat_ws` function. ({issue}`4680`)
+* Add support for {func}`extract` for `time with time zone` values with precision other than 3. ({issue}`5539`)
+* Add dynamic filtering support for right joins. ({issue}`5555`) 
+* Add support for catalog session property to file based session property manager. ({issue}`5330`)
+* Add support for catalog session property to database based session property manager. ({issue}`5330`)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-345.md:9]: Maybe merge those (as file and database are only session property managers we support right now).

```
Add support for catalog session properties to session property manager. Both file and database backed property managers are supported.
```

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+* Improve query scalability by increasing broadcast join buffer size to 200MB.
+  Broadcast join buffer size can be set using `sink.max-broadcast-buffer-size` feature config. ({issue}`5551`)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-345.md:11]:
```suggestion
* Improve query scalability by increasing default broadcast join buffer size to 200MB.
  Broadcast join buffer size can be set using `sink.max-broadcast-buffer-size` feature config. ({issue}`5551`)
```

File: presto-docs/src/main/sphinx/release/release-345.md (hunk 1 above)
Comment (losipiuk) [reply presto-docs/src/main/sphinx/release/release-345.md:26]: Add/Fix/Improve. I think we should keep each point a sentence in present tense.

File: presto-docs/src/main/sphinx/release/release-345.md (hunk 3 above)
Comment (electrum) [reply presto-docs/src/main/sphinx/release/release-345.md:40]: I think it's fine to describe in English

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+* Add column security, column mask and row filter to file-based access controls. ({issue}`5460`)
+* Enforce access control for column references in `USING` clause. ({issue}`5620`)
+
+## Hive Connector Changes
+
+* Use `hive.domain-compaction-threshold` threshold for compacting dynamic filters. ({issue}`5365`)
+* Support reading Parquet timestamp encoded as microseconds. ({issue}`5483`)
+* Fix failure when reading Parquet `timestamp` columns encoded as `int64`. ({issue}`5443`)
+* Fixes disk space accounting when cachin is enabled. ({issue}`5621`)
+* Improve cached files distribution. ({issue}`5621`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-345.md:34]: How about

* Improve storage caching by better distributing files across workers.

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  setting the `jwt.key-file` configuration property to a `http` or `https` url. ({issue}`5419`)
+* Add column security, column mask and row filter to file-based access controls. ({issue}`5460`)
+* Enforce access control for column references in `USING` clause. ({issue}`5620`)
+
+## Hive Connector Changes
+
+* Use `hive.domain-compaction-threshold` threshold for compacting dynamic filters. ({issue}`5365`)
+* Support reading Parquet timestamp encoded as microseconds. ({issue}`5483`)
+* Fix failure when reading Parquet `timestamp` columns encoded as `int64`. ({issue}`5443`)
+* Fixes disk space accounting when cachin is enabled. ({issue}`5621`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-345.md]: * Fix disk space accounting for storage caching.

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
+# Release 345 (XX Oct 2020)
+
+## General Changes
+
+* Add {func}`concat_ws` function. ({issue}`4680`)
+* Add support for {func}`extract` for `time with time zone` values with precision other than 3. ({issue}`5539`)
+* Add dynamic filtering support for right joins. ({issue}`5555`) 
+* Add support for catalog session properties to session property manager. Both file and database backed property
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-345.md:8]: This feature already existed in the engine, so we should say

* Add support for catalog session properties to the file and database backed session property managers.

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+## General Changes
+
+* Add {func}`concat_ws` function. ({issue}`4680`)
+* Add support for {func}`extract` for `time with time zone` values with precision other than 3. ({issue}`5539`)
+* Add dynamic filtering support for right joins. ({issue}`5555`) 
+* Add support for catalog session properties to session property manager. Both file and database backed property 
+  managers are supported. ({issue}`5330`)
+* Improve query scalability by increasing the default broadcast join buffer size to 200MB.
+  Broadcast join buffer size can be set using `sink.max-broadcast-buffer-size` feature config. ({issue}`5551`)
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-345.md:11]: using the `xxx` configuration property.

**Comment by martint** [comment]
Updated

**Review by losipiuk — APPROVED** [review]

File: presto-docs/src/main/sphinx/release/release-345.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
+# Release 345 (XX Oct 2020)
```
Comment (losipiuk) [inline presto-docs/src/main/sphinx/release/release-345.md:1]: TODO

**Review by phd3 — APPROVED** [review]
looks good, not sure if the date changes through some other mechanism.

**Review by dain — APPROVED** [review]

**Review by ebyhr — APPROVED** [review]

**Review by Praveen2112 — APPROVED** [review]

**Pull request closed without merging**